src/SDKs/Azure.Base/data-plane/Azure.Base/Diagnostics/HttpPipelineEventSource.cs

```diff
@@ -3,77 +3,191 @@
+        }
 
         [NonEvent]
         public void ResponseDelay(HttpPipelineResponse response, long delayMilliseconds)
-            => ResponseDelayCore(delayMilliseconds);
+        {
+            ResponseDelayCore(delayMilliseconds);
+        }
 
         // TODO (pri 2): there are more attribute properties we might want to set
-        [Event(LOG_REQUEST, Level = EventLevel.Informational)]
-        void ProcessingRequest(string request)
+        [Event(RequestEvent, Level = EventLevel.Informational)]
+        private void Request(string correlationId, string method, string uri, string headers)
         {
-            // TODO (pri 2): is EventKeywords.None ok?
-            if (IsEnabled(EventLevel.Informational, EventKeywords.None)) {
-                WriteEvent(LOG_REQUEST, request);
+            WriteEvent(RequestEvent, correlationId, method, uri, headers);
+        }
+
+        [Event(RequestContentEvent, Level = EventLevel.Verbose)]
+        private void RequestContent(string correlationId, byte[] content)
+        {
+            WriteEvent(RequestContentEvent, correlationId, content);
+        }
+
+        [Event(ResponseEvent, Level = EventLevel.Informational)]
+        private void Response(string correlationId, int status, string headers)
+        {
+            WriteEvent(ResponseEvent, correlationId, status, headers);
+        }
+
+
+        [Event(ResponseContentEvent, Level = EventLevel.Verbose)]
+        private void ResponseContent(string correlationId, byte[] content)
+        {
+            WriteEvent(ResponseContentEvent, correlationId, content);
+        }
+
+        [Event(ErrorResponseEvent, Level = EventLevel.Error)]
+        public void ErrorResponse(string correlationId, int status, string headers)
+        {
+            WriteEvent(ErrorResponseEvent, correlationId, status, headers);
+        }
+
+        [Event(ErrorResponseContentEvent, Level = EventLevel.Informational)]
+        private void ErrorResponseContent(string correlationId, byte[] content)
+        {
+            WriteEvent(ErrorResponseContentEvent, correlationId, content);
+        }
+
+        [Event(RequestDelayEvent, Level = EventLevel.Warning)]
+        private void ResponseDelayCore(long delayMilliseconds)
+        {
+            if (IsEnabled(EventLevel.Warning, EventKeywords.None))
+            {
+                WriteEvent(RequestDelayEvent, delayMilliseconds);
             }
         }
 
-        [Event(LOG_RESPONSE, Level = EventLevel.Informational)]
-        void ProcessingResponse(string response)
+        private static async Task<byte[]> FormatContentAsync(Stream responseContent)
         {
-            if (IsEnabled(EventLevel.Informational, EventKeywords.None)) {
-                WriteEvent(LOG_RESPONSE, response);
+            using (var memoryStream = new MemoryStream())
+            {
+                await responseContent.CopyToAsync(memoryStream);
+
+                // Rewind the stream
+                responseContent.Position = 0;
```

Comment:
Should we check `CanSeek` before setting the position since this takes a base `Stream` instead of something specialized?   Perhaps, it may be worth considering changing the type to something more specialized that we're expecting?

Reply:
I'm not sure if we can reasonably support non-seekable response streams given our overall requirements.

> Should we check CanSeek before setting the position since this takes a base Stream instead of something specialized

If the stream is not seekable and we've read the body we won't be able to deserialize the response at the end of the pipeline.

Another solution to this is wrapping the response stream into a logging stream so data can be logged while it's being read but it's way more complicated.